guides/source/action_view_helpers.md

```diff
@@ -5,668 +5,1054 @@ Action View Helpers
 ```
 
-See the [API Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/NumberHelper.html#method-i-number_to_human_size) for more information.
+```html
+<script type="application/javascript">
+//<![CDATA[
+alert('All is good')
+//]]>
+</script>
+```
 
-#### number_to_percentage
+Instead of passing the content as an argument, you can also use a block.
 
-Formats a number as a percentage string.
+```html+erb
+<%= javascript_tag type: "application/javascript" do %>
+  alert("Welcome to my app!")
+<% end %>
+```
+
+See the [API
+Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/JavaScriptHelper.html#method-i-javascript_tag)
+for more information.
+
+Alternative Tags
+----------------------------------------------
+
+A set of methods to generate HTML tags programmatically.
+
+### tag
+
+Generates a standalone HTML tag with the given `name` and `options`.
+
+Every tag can be built with:
 
 ```ruby
-number_to_percentage(100, precision: 0) # => 100%
+tag.some_tag_name(optional content, options)
 ```
 
-See the [API Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/NumberHelper.html#method-i-number_to_percentage) for more information.
+where tag name can be e.g. `br`, `div`, `section`, `article`, or any tag really.
 
-#### number_to_phone
-
-Formats a number into a phone number (US by default).
+For example, here are some common uses:
 
 ```ruby
-number_to_phone(1235551234) # => 123-555-1234
+tag.h1 "All titles fit to print"
+# => <h1>All titles fit to print</h1>
+
+tag.div "Hello, world!"
+# => <div>Hello, world!</div>
 ```
 
-See the [API Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/NumberHelper.html#method-i-number_to_phone) for more information.
+#### Options
 
-#### number_with_delimiter
+You can use symboled key options to add attributes to the generated tag.
 
-Formats a number with grouped thousands using a delimiter.
+```ruby
+tag.section class: %w( kitties puppies )
+# => <section class="kitties puppies"></section>
+```
+
+In addition, HTML `data-*` attributes can be passed to the `tag` helper as a key value pair, where the key is `data` and the value is a hash containing sub-attributes. The sub-attributes are then converted to `data-*` attributes that are dasherized in order to work well with JavaScript.
 
 ```ruby
-number_with_delimiter(12345678) # => 12,345,678
+tag.div data: { user_id: 123 }
+# => <div data-user-id="123"></div>
 ```
 
-See the [API Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/NumberHelper.html#method-i-number_with_delimiter) for more information.
+See the [API
+Documentation](https://api.rubyonrails.org/classes/ActionView/Helpers/TagHelper.html#method-i-tag)
+for more information.
 
-#### number_with_precision
+### token_list
 
-Formats a number with the specified level of `precision`, which defaults to 3.
+Returns a string of tokens built from the `args` provided. This method is also
+aliased as `class_names`.
```

Comment:
Since the argument name is not as relevant in this case, when compared to a few others like `excerpt` that have more arguments with more meaningful naming, maybe we can just use "arguments" more generically?

```suggestion
Returns a string of tokens built from the arguments provided. This method is
also aliased as `class_names`.
```